ailment/expression.py

```diff
@@ -29,7 +29,7 @@ def has_atom(self, atom, identity=True):
             return self.likes(atom)
 
     def __eq__(self, other):
-        return self.likes(other) and self.idx == other.idx
+        return self.likes(other)
```

Comment:
I believe you are just making `likes == __eq__`, which negates the reason why there is a `likes()` in the first place...